### PR TITLE
Add post install script to download solc

### PR DESCRIPTION
**Pull request by eggplantzzz**
This is the first of a series of PRs that will be made in order to remove solc from Truffle's dependencies.

In this installment, a `--download-solc` flag was added to the compile command. It takes a solc version as an argument and uses the `CompilerSupplier` to download and cache that version of the solidity compiler.

In addition, a post install script has been added to the package.json that downloads the "default" version of the solidity compiler that eventually will be used to replace the bundled version.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "build:truffle-contract-schema": "cd packages/truffle-contract-schema && npm run build",
     "lerna:bootstrap": "lerna bootstrap",
     "bootstrap": "./scripts/bootstrap.sh",
+    "postinstall": "./scripts/downloadSolc.sh",
```
Comment (gnidan) [inline package.json]: Can we just inline the script here? I'm worried about non-UNIX operating systems where that file might not be compatible.

Should be fine if the entirety of the `postinstall` is just `truffle compile --download-solc 0.5.4`

If it turns into more than one line later, where a shell script becomes more warranted, we'll have to look into cross-platform solutions then.

Comment (gnidan) [inline package.json]: Can we just inline the script here? I'm worried about non-UNIX operating systems where that file might not be compatible.

Should be fine if the entirety of the `postinstall` is just `truffle compile --download-solc 0.5.4`

If it turns into more than one line later, where a shell script becomes more warranted, we'll have to look into cross-platform solutions then.

File: scripts/downloadSolc.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+truffle compile --download-solc 0.5.4
```
Comment (gnidan) [inline scripts/downloadSolc.sh]: (Note from above: let's remove this file and inline it into the package.json)

File: package.json (hunk 1 above)
Comment (eggplantzzz) [reply package.json]: Sounds fine to me

**Pull request merged**